src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/edit-video-unit/edit-video-unit.component.ts

```diff
@@ -34,37 +41,51 @@ export class EditVideoUnitComponent implements OnInit {
             .pipe(
                 take(1),
                 switchMap(([params, parentParams]) => {
-                    const videoUnitId = Number(params.get('videoUnitId'));
+                    this.videoUnitId = Number(params.get('videoUnitId'));
                     this.lectureId = Number(parentParams.get('lectureId'));
-                    return this.videoUnitService.findById(videoUnitId, this.lectureId);
+                    const videoUnitObservable = this.videoUnitService.findById(this.videoUnitId, this.lectureId);
+                    const lectureObservable = this.lectureService.findWithDetails(this.lectureId);
```

Comment:
why do we need a new call rather than returning video unit & lecture via one call?